scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+                #End if
+
+                fils = sorted(list(Path(case_ts_locs[case_idx]).glob(f"*{var}.*.nc")))
+                ts_ds = _load_dataset(fils)
+
+                #Check if variable has a vertical coordinate:
+                if 'lev' in ts_ds.coords or 'ilev' in ts_ds.coords:
+                    print(f"\t   Variable '{var}' has a vertical dimension, "+\
+                        "which is currently not supported for the time series plot. Skipping...")
+                    #Skip this variable and move to the next variable in var_list:
+                    continue
+                avg_case,_,yrs_case,unit = _data_calcs(var,ts_ds=ts_ds,subset=None)
+
+            #End if (RESTOM)
+
+            #Get yearly averages for all available years
+            vals_case = [avg_case.sel(time=i).mean() for i in yrs_case]
+
+            #Grab min and max vals from each test case
+            mins.append(np.nanmin(vals_case))
+            maxs.append(np.nanmax(vals_case))
+
+            #Get int of years for plotting on x-axis
+            yrs_case_int = yrs_case.astype(int)
+
+            name = all_nicknames[case_idx]
+            if case_idx == (len(all_case_names)-1):
+                name = f"{name} (baseline)"
+
+            #Add case to plot (ax)
+            ax.plot(yrs_case_int, vals_case, color_dict["marker"], c=color_dict["color"],label=name)
+
+            #For the minor ticks, use no labels; default NullFormatter.
+            ax.tick_params(which='major', length=7)
+            ax.tick_params(which='minor', length=5)
+        #End for (case names)
+
+        #Set Main title for subplots:
+        ax.set_title(f"Time Series {title_var}: {var} - {season}",loc="left")
+        
+        if rolling:
+            ax.set_title(f"5-yr rolling average",loc="right")
+
+        #Minor tweak to not plot variables that have vertical levels.
+        #TODO: Clean up this check - JR
+        #mins and maxs are blank lists when trying to ignore variables...
+        if (mins) or (maxs):
+            ax = _format_yaxis(ax, case_num, unit, **vres)
+            ax = _format_xaxis(ax, yrs)
+
+            #Set up legend
+            fig = _make_fig_legend(case_num, fig)
+
+            #Save plot
+            plot_name = plot_loc / f"{var}_{season}_TimeSeries_Mean.{plot_type}"
+            plt.savefig(plot_name, facecolor='w')
+            plt.close()
+
+            #Add plot to website (if enabled):
+            adfobj.add_website_data(plot_name, var, case,
+                                    season=season,
+                                    plot_type="TimeSeries",
+                                    multi_case=multi_case)
+        #End if (min/max)
+
+    #End for (vars)
+    #End ANN plots
+
+    #Seasonally weighted plots
+    # - DJF, MAM, JJA, SON
+    ##########################
+    print("\n  Generating time series for seasonally weighted...")
+    for var in ts_var_list_s:
```

Comment:
It feels like a lot of the code inside this loop is the same as the `ANN` code above.  Maybe it would be worth making a Github issue so that in the future one of us can remember to try and find a clever way to combine the two code blocks?

Reply:
Yeah, I gave it a couple of tries before and was unsuccessful. Maybe if/when you get some time you can see if something can make sense. I'll also give it another go soon since I've had some time away from it now